main/src/io/github/iltotore/iron/RefinedTypeOps.scala

```diff
@@ -0,0 +1,18 @@
+package io.github.iltotore.iron
+
+type RefinedTypeOps[T] = T match
+  case IronType[a, c] => RefinedTypeOpsImpl[a, c, T]
+
+class RefinedTypeOpsImpl[A, C, T]:
+  inline def apply(value: A)(using Constraint[A, C]): T =
```

Comment:
I'm realizing there will be a problem with already refined types. Does the following code compiles?

```scala
opaque type Temperature = Int :| Positive
object Temperature extends RefinedTypeOps[Temperature]
```
```
val x: Int :| Positive = 5
val t = Temperature(5)
```

Reply:
Explicit function (for `IronType`) was required, otherwise the error says `Note: Due to a Scala limitation, already-refined types cannot be tested at compile-time (unless proven by an Implication).`
```
  inline def fromRefined(value: IronType[A, C])(using Constraint[A, C]): T =
    value.asInstanceOf[T]
```

```
    test("re-eval") {
      val x: Double :| Positive = 5
      val t = Temperature.fromRefined(x)
    }
```
Does additional function work?

Reply:
I think an additional method is the best choice. However the signature should probably look like this:

```scala
inline def fromRefined[C1](value: IronType[A, C])(using C1 ==> C): T =
  value.asInstanceOf[T]
```

This way, this should work:

```scala
val x: Int :| Positive = ???
val y: Int :| Greater[10]

val t1 = Temperature.fromRefined(x) //Compiles
val t2 = Temperature.fromRefined(y) //Also compiles
```

Also I would rename `fromRefined` to something like `fromIronType` or `fromIron` for two reasons:
- Refined types are already called "Iron types" in Iron
- IMHO it sounds like "from refined types of refined (the lib)"

Reply:
Makes sense. Emphasis here was "is it ok to have another function (instead of doing some implicit conversion from A to A |: C to let apply consume everything)?"
Going to take `fromIronType`

Reply:
> using C1 ==> C

Thanks for head-ups

Reply:
@Iltotore 
Seems like `Positive` constraint automatically turns constraint into "integer based".

The function:
```
inline def fromIronType[C1](value: IronType[A, C1])(using Implication[C1, C]): T =
  value.asInstanceOf[T]
```

Int based types work ok:
```
opaque type Temperature = Int :| Positive
object Temperature extends RefinedTypeOps[Temperature]

```scala
    test("re-eval") {
      val x: Int :| Positive = 5
      val y: Int :| Greater[10] = 15
      val t1 = Temperature.fromIronType(x)
      val t2 = Temperature.fromIronType(y)

      assert(t1 == Temperature(5), "should be result of 'apply'")
      assert(t2 == Temperature(15), "should be result of 'apply'")
    }
```


But for double-based type there is a different picture:
```
opaque type Temperature = Double :| Positive
object Temperature extends RefinedTypeOps[Temperature]


    test("re-eval") {
      val x: Double :| Positive = 5.0
      val y: Double :| Greater[10.0] = 15.0
      val t1 = Temperature.fromIronType(x)
      val t2 = Temperature.fromIronType(y) //DOESN'T COMPILE

      assert(t1 == Temperature(5.0), "should be result of 'apply'")
      assert(t2 == Temperature(15.0), "should be result of 'apply'")
    }
```

```
Could not prove that io.github.iltotore.iron.constraint.numeric.Greater[(10.0d : Double)] implies io.github.iltotore.iron.constraint.any.DescribedAs[
  io.github.iltotore.iron.constraint.numeric.Greater[(0 : Int)]
, ("Should be strictly positive" : String)].
```

It seems more like explicit bug that should be tackled as explicit implication tuning but please let me know if I am missing something.

Reply:
I think I found the cause of this issue. Does declaring `y` as a `Double :| Greater[10]` works?

Reply:
Yes, `val y: Double :| Greater[10] = 15.0` (not `10.0`) works ok

Reply:
This issue does not come from your PR. I will work on it separately.